File: java_api/src/main/java/globalincidents/controller/GetItemController.java
package globalincidents.controller;

import org.json.JSONArray;
import org.json.JSONObject;
import org.springframework.util.MultiValueMap;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;
import utils.ApiUtils;
import utils.Constants;
import utils.DBConnection;

@RequestMapping(value = "/get-item")
@RestController
public class GetItemController extends BaseController{
  private int mId = (Integer) Constants.API_ID.getDefault();

  @Override
  public String run(@RequestParam MultiValueMap<String,String> params) {
    if(this.mId == -1)
    {
      JSONObject obj = new JSONObject();
      obj.put("message", "Some of the mandatory parameters are missing. Please consult our API documentation");

      return obj.toString();
    }

    JSONArray toReturn = DBConnection.ExecuteQuery("select * from incidents where id=" + this.mId);
    return toReturn.toString();
  }

  @Override
  public void getParams(MultiValueMap<String, String> params) {
    this.mId = Integer.parseInt((String)ApiUtils.getParamString(Constants.API_ID, params));
  }
}

File: kotlin_api/src/main/kotlin/globalincidents/controller/GetItemController.kt
package globalincidents.controller

import org.json.JSONObject
import org.springframework.util.MultiValueMap
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController
import utils.ApiUtils
import utils.Constants
import utils.DBConnection

@RestController
class GetItemController : BaseController() {
    internal var mId: Any? = Constants.API_ID.getDefault()

    @GetMapping("/get-item")
    override fun execute(@RequestParam params: MultiValueMap<String, String>): String {
        super.execute(params)

        if(this.mId == -1)
        {
            val obj = JSONObject()
            obj.put("message", "Some of the mandatory parameters are missing. Please consult our API documentation")

            return obj.toString()
        }

        val toReturn = DBConnection.executeQuery("select * from incidents where id=" + this.mId)
        return toReturn.toString()
    }

    override fun getParams(params: MultiValueMap<String, String>) {
        this.mId = ApiUtils.getParam(Constants.API_ID, params)
    }
}